OpenRA.Mods.Common/Traits/World/BuildableTerrainOverlay.cs

```diff
@@ -0,0 +1,83 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	class BuildableTerrainOverlayInfo : TraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference(typeof(BuildingInfo))]
+		public readonly string CheckOnBuilding = null;
```

Comment:
Did you have a specific reason for routing this though an actor instead of directly listing terrain types?

Reply:
Maintenance maybe, but that quickly falls apart. I removed it.